bench/Autofac.Benchmarks/Decorators/DecoratorBenchmarkBase.cs

```diff
@@ -9,17 +9,33 @@ public abstract class DecoratorBenchmarkBase<TCommandHandler>
         protected IContainer Container { get; set; }
 
         [Benchmark]
-        public virtual void EnumerableResolve()
+        [Arguments(1)]
+        [Arguments(2)]
+        [Arguments(3)]
+        public virtual void ResolveEnumerableT(int repetitions)
         {
-            var item = this.Container.Resolve<IEnumerable<TCommandHandler>>();
-            GC.KeepAlive(item);
+            var iteration = 1;
+            object item = null;
+            while(iteration++ < repetitions)
```

Comment:
This logic means that if only 1 repetition is specified, no work is done; did you mean ``<=``?.

Same for the ``ResolveT``.

That explains why there is such a significant discrepancy between the first and second repetitions.